R/utils.R

```diff
@@ -220,6 +220,87 @@ check_filter_datanames <- function(filters, datanames) {
   }
 }
 
+
+#' Extract datasets for modules
+#'
+#' @param data (`teal_data`)
+#' @param modules (`teal_modules`) object
+#' @param filters (`teal_slices`) object
+#' @param filtered_data_singleton A result of `teal_data_to_filtered_data` applied to `data`.
+#' @return Returns list of same shape as `modules`, containing `FilteredData` at every leaf.
+#' If module specific, each leaf contains different instance, otherwise every leaf contains `filtered_data_singleton`.
+#' @keywords internal
+modules_datasets <- function(data, modules, filters, filtered_data_singleton = teal_data_to_filtered_data(data)) {
+  checkmate::assert_class(data, "teal_data")
+  checkmate::assert_multi_class(modules, c("teal_modules", "teal_module"))
+  checkmate::assert_class(filters, "modules_teal_slices")
+  checkmate::assert_r6(filtered_data_singleton, "FilteredData")
+
+  if (!isTRUE(attr(filters, "module_specific"))) {
+    slices <- shiny::isolate({
+      Filter(function(x) x$id %in% attr(filters, "mapping")$global_filters, filters)
+    })
+    filtered_data_singleton$set_filter_state(slices)
+    return(modules_structure(modules, filtered_data_singleton))
+  }
+
+  if (inherits(modules, "teal_module")) {
+    # 1. get datanames
+    datanames <-
+      if (is.null(modules$datanames) || identical(modules$datanames, "all")) {
+        include_parent_datanames(
+          teal_data_datanames(data),
+          teal.data::join_keys(data)
+        )
+      } else {
+        modules$datanames
+      }
+    # 2. subset filters (global + dedicated)
+    slices <- shiny::isolate({
+      Filter(x = filters, f = function(x) {
+        x$dataname %in% datanames &&
+          (x$id %in% attr(filters, "mapping")$global_filters ||
+            x$id %in% unique(unlist(attr(filters, "mapping")[modules$label]))) # nolint
```

Comment:
Please specify linter exception.

Reply:
this is this ridiculous styler vs lintr infinite war : D

Reply:
I don't want an explanation, I want a specification. Which linter was turned off for this line?

Reply:
got it. was trying to turn off `indentation_linter` https://github.com/insightsengineering/teal/pull/1118
put a PR for this

Reply:
pertect